corehq/apps/registration/utils.py

```diff
@@ -179,7 +180,8 @@ def request_new_domain(request, form, org, domain_type=None, new_user=True):
         dom_req.activation_guid = uuid.uuid1().hex
 
     new_domain = Domain(
-        name=form.cleaned_data['domain_name'],
+        name='',
```

Comment:
Jenny's comment "@snopoke This actually was deliberate: users no longer enter their own names for domains, so the domain is just created with a blank string, which tells the constructor to use the id for a name"